slack_bolt/workflows/step/step.py

```diff
@@ -1,19 +1,263 @@
+        middleware: Optional[Union[Callable, Middleware]] = None,
+        lazy: Optional[List[Callable[..., None]]] = None,
+    ):
+        """Register a new execute listener with details.
+        You can use this method as decorator as well.
+
+        @my_step.execute
+        def execute_my_step(step, complete, fail):
+            pass
+
+        It's also possible to add additional listener matchers and/or middleware
+
+        @my_step.save(matchers=[is_valid], middleware=[update_context])
+        def execute_my_step(step, complete, fail):
+            pass
+        """
+
+        if len(args) == 1:
+            func = args[0]
+            self._execute = self._to_listener("execute", func, matchers, middleware)
+            return func
+
+        def _inner(func):
+            functions = [func] + (lazy if lazy is not None else [])
+            self._execute = self._to_listener(
+                "execute", functions, matchers, middleware
+            )
+
+            @wraps(func)
+            def _wrapper(*args, **kwargs):
+                return func(*args, **kwargs)
+
+            return _wrapper
+
+        return _inner
+
+    def build(self) -> "WorkflowStep":
+        """Constructs a WorkflowStep object. This method may raise an exception
+        if the builder doesn't have enough configurations to build the object.
+
+        :return: WorkflowStep object
+        """
+        if self._edit is None:
+            raise BoltError(f"edit listener is not registered")
+        if self._save is None:
+            raise BoltError(f"save listener is not registered")
+        if self._execute is None:
+            raise BoltError(f"execute listener is not registered")
+
+        return WorkflowStep(
+            callback_id=self.callback_id,
+            edit=self._edit,
+            save=self._save,
+            execute=self._execute,
+            app_name=self.app_name,
+        )
+
+    # ---------------------------------------
+
+    def _to_listener(
+        self,
+        name: str,
+        listener_or_functions: Union[Listener, Callable, List[Callable]],
+        matchers: Optional[Union[Callable[..., bool], ListenerMatcher]] = None,
+        middleware: Optional[Union[Callable, Middleware]] = None,
+    ) -> Listener:
+        return WorkflowStep.build_listener(
+            callback_id=self.callback_id,
+            app_name=self.app_name,
+            listener_or_functions=listener_or_functions,
+            name=name,
+            matchers=self.to_listener_matchers(self.app_name, matchers),
+            middleware=self.to_listener_middleware(self.app_name, middleware),
+        )
+
+    @staticmethod
+    def _to_listener_functions(
+        kwargs: dict,
+    ) -> Optional[List[Callable[..., Optional[BoltResponse]]]]:
+        if kwargs:
+            functions = [kwargs["ack"]]
+            for sub in kwargs["lazy"]:
+                functions.append(sub)
+            return functions
+        return None
```

Comment:
Is there any problem if we return `[] `instead of `None`?

Reply:
Indeed. I agree we can return an empty list in the case 👍